rest_framework_dataclasses/typing_utils.py

```diff
@@ -26,6 +26,12 @@
 else:
     GenericAlias = typing._GenericAlias
 
+# typing.UnionType was added in Python 3.10 for new PEP 604 pipe union syntax
+try:
+    from types import UnionType
+except ImportError:
+    UnionType = None
```

Comment:
My other thought was
```python
try:
     from types import UnionType
     UNION_TYPES = (typing.Union, UnionType)
except ImportError:
     UNION_TYPES = (typing.Union,)
```
And then check if `origin in UNION_TYPES`

Reply:
Yes, I'd prefer that.